WcaOnRails/app/webpacker/components/EditSchedule/store/reducer.js

```diff
@@ -0,0 +1,216 @@
+        ...venue,
+        rooms: venue.rooms.map((room) => ({
+          ...room,
+          activities: room.activities.map((activity) => (
+            activity.id === payload.activityId
+              ? scaleActivitiesByDuration(activity, payload.isoDeltaStart, payload.isoDeltaEnd)
+              : activity
+          )),
+        })),
+      })),
+    },
+  }),
+
+  [EditVenue]: (state, { payload }) => ({
+    ...state,
+    wcifSchedule: {
+      ...state.wcifSchedule,
+      venues: state.wcifSchedule.venues.map((venue) => (venue.id === payload.venueId ? {
+        ...venue,
+        [payload.propertyKey]: payload.newProperty,
+        rooms: venue.rooms.map((room) => ({
+          ...room,
+          activities: (
+            payload.propertyKey === 'timezone'
+              ? room.activities.map((activity) => (
+                changeActivityTimezone(
+                  activity,
+                  venue.timezone,
+                  payload.newProperty,
+                )
+              ))
+              : room.activities
+          ),
+        })),
+      } : venue)),
+    },
+  }),
+
+  [EditRoom]: (state, { payload }) => ({
+    ...state,
+    wcifSchedule: {
+      ...state.wcifSchedule,
+      venues: state.wcifSchedule.venues.map((venue) => ({
+        ...venue,
+        rooms: venue.rooms.map((room) => (room.id === payload.roomId ? {
+          ...room,
+          [payload.propertyKey]: payload.newProperty,
+        } : room)),
+      })),
+    },
+  }),
+
+  [RemoveVenue]: (state, { payload }) => ({
+    ...state,
+    wcifSchedule: {
+      ...state.wcifSchedule,
+      venues: state.wcifSchedule.venues.filter((venue) => venue.id !== payload.venueId),
+    },
+  }),
+
+  [RemoveRoom]: (state, { payload }) => ({
+    ...state,
+    wcifSchedule: {
+      ...state.wcifSchedule,
+      venues: state.wcifSchedule.venues.map((venue) => ({
+        ...venue,
+        rooms: venue.rooms.filter((room) => room.id !== payload.roomId),
+      })),
+    },
+  }),
+
+  [AddVenue]: (state) => ({
+    ...state,
+    wcifSchedule: {
+      ...state.wcifSchedule,
+      venues: [
+        ...state.wcifSchedule.venues,
+        {
+          id: nextVenueId(state.wcifSchedule),
+          latitudeMicrodegrees: 0,
+          longitudeMicrodegrees: 0,
+          rooms: [],
+          extensions: [],
+        },
+      ],
+    },
+  }),
+
+  [AddRoom]: (state, { payload }) => ({
+    ...state,
+    wcifSchedule: {
+      ...state.wcifSchedule,
+      venues: state.wcifSchedule.venues.map((venue) => (venue.id === payload.venueId ? {
+        ...venue,
+        rooms: [
+          ...venue.rooms,
+          {
+            id: nextRoomId(state.wcifSchedule),
+            activities: [],
+            extensions: [],
+          },
+        ],
+      } : venue)),
+    },
+  }),
+};
+
+export default function rootReducer(state, action) {
+  const reducer = reducers[action.type];
+  if (reducer) {
+    return reducer(state, action);
+  }
+  return state;
```

Comment:
Do we want to log some kind of error or warning here about reducer case not found?

Reply:
That'd be helpful for debugging. Ideally there shouldn't be errors when this is tested and pushed to prod.

Reply:
To be honest, I blatantly stole this from `editEvents`, which doesn't have dedicated error handling either.
Rather than introducing different approaches here and there, I'd prefer having a unified approach with react-redux (or similar) in a separate PR.